core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -1047,6 +1047,14 @@ package object config {
       .checkValue(v => v > 0, "The value should be a positive integer.")
       .createWithDefault(2000)
 
+  private[spark] val SHUFFLE_USE_OLD_FETCH_PROTOCOL =
```

Comment:
What happens if you connect to an old shuffle service without setting this? Will things just fail (and always fail)? 

Probably ok if they do. Although a more user-friendly error, if possible, might be good.

Reply:
```
What happens if you connect to an old shuffle service without setting this? Will things just fail (and always fail)?
```
We'll always fail with the UnsupportedOperationException:`Unexpected message: FetchShuffleBlocks`. This work is done in #24565.

```
Although a more user-friendly error, if possible, might be good.
```
Yeah, in this PR we had detailed log in DAGScheduler.
Let me think how to add a more user-friendly error in the follow-up work for the old shuffle service, currently, we only have some doc in the migration guide. https://github.com/apache/spark/pull/24565/files#diff-3f19ec3d15dcd8cd42bb25dde1c5c1a9R139